### DISCUSSION TITLE
Questions about firmware requirement in README

### DISCUSSION THREAD

**Discussion by hzulla** [wlcrs/huawei_solar#693] [Setting up the integration] [answered]
Hi there,

(forgive me, I'm new to this.)

I recently purchased a small solar panel array and had it installed. My devices are:

- Inverter: SUN2000-5KTL-M1 / software version V100R001C00SPC162
- Dongle: SDongleA-05 / software version V100R001C00SPC133

The home network is connected via the ethernet interface, WiFi isn't used.

Modbus-TCP isn't activated yet. I do not have an installer account myself, but contacted the craftsman who installed the system to have Modbus-TCP activated.

The [README](https://github.com/wlcrs/huawei_solar?tab=readme-ov-file#prerequisites) for the huawei_solar integration states

> "This integration supports inverters running **V200**R001C00 firmware. If you are running an older V100R001C00 firmware, then please consult the "Upgrade guide" included with V200R001C00 firmware versions to check if your inverter is eligibile for an upgrade."

I have tried to find the mentioned "Upgrade guide" but couldn't. The Huawei support web pages are confusingly designed and I failed to find the information if there is a V200 firmware available for my devices. While I found firmware downloads, those weren't freely downloadable.

How can I find out if there are V200 firmware upgrades for my devices?

Will the integration fail to work with the V100 firmware version on my devices or will it just result in limited functionality?

Thanks!

**Comment by hzulla** [comment]
Also, do I need a V200 firmware for both devices or just for the dongle?

**Comment by hzulla** [comment]
Update: So my home assistant was now able to connect the integration with the dongle/inverter, despite both having V100 firmwares. Do I have to expect any problems with this non-V200 setup?

**Comment by achtbitunix** [comment]
Hi @hzulla 

I'm running V100R for nearly a year and did not have any problems.
Maybe because I'm a bit conservative regarding updates ;)

Inverter | V100R001C00SPC156
Dongle | V100R001C00SPC133

**Reply by Roving-Ronin** [reply, marked as answer]
@hzulla 

Refer to the Wiki where co formed firmware is listed.

https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware-versions#three-phase---sun2000-m1-hybrid

You'll note there that v100xxx firmware is the current naming format for the inverters, with iirc SPC162 being released in the last week or two.

Conversely v200xxx firmware is the newer branch of firmware for the sdongle, with SPC114 being the latest / newer versions for it. The v100xxx you have on the dongle will work, but v200xxx is needed at some point if you intend to leverage some of the newer products being released and/or new patches for the inverters/batteries.

**Comment by hzulla** [comment]
Update: While the connection worked with the V100 firmware, my installer updated the firmware to V200 upon my request - and it still works.

Thanks to everyone involved!
